dot/state/block.go

```diff
@@ -236,7 +238,7 @@ func (bs *BlockState) HasHeader(hash common.Hash) (bool, error) {
 
 // GetHeader returns a BlockHeader for a given hash
 func (bs *BlockState) GetHeader(hash common.Hash) (*types.Header, error) {
-	result := new(types.Header)
+	result := types.NewEmptyHeader()
```

Comment:
If I'm not mistaken you would need the custom VDTs for that. Since the header.Digest is a vdt slice, we still need to specify what VDT items to unmarshall into. Like I think unmarshalling into types.header would give you an index not found error since it scale doesn't know what its unmarshalling.

Reply:
oh I see.  Yea makes sense.